### PR TITLE
Xcode7

### PR DESCRIPTION
**Pull request by codecaffeine**
Updating to Xcode7. Most of the file changes are due to running `pod install` with the latest cocoapods.

The only effective code change is in _URBNArrayDataSourceAdapter.m_ where I’ve added the following `super init` override:

```objc
- (instancetype)init {
    return [self initWithItems:@[]];
}
```

Where it will init with no items if the generic init method is called.

### CODE REVIEW

**Comment by codecaffeine** [comment]
:-( fail?

**Comment by ndis1** [comment]
@codecaffeine try setting the xcode version travis uses in travis.yml

**Comment by codecaffeine** [comment]
:warning: dont merge yet

**Comment by codecaffeine** [comment]
Alright @jgrandelli, now check it out

File: Pod/Classes/URBNDataSourceAdapter.h [hunk 1]
```diff
@@ -9,16 +9,18 @@
...
+NS_ASSUME_NONNULL_BEGIN
+
 typedef NS_ENUM(NSUInteger, URBNSupplementaryViewType) {
     URBNSupplementaryViewTypeHeader,
     URBNSupplementaryViewTypeFooter
 };
 
-typedef NSString *(^URBNSupplementaryViewReuseIdentifierBlock) (NSString *kind, NSIndexPath *indexPath);
-typedef NSString *(^URBNReuseableIdentifierBlock) (id item, NSIndexPath *indexPath);
+typedef NSString * _Nonnull (^URBNSupplementaryViewReuseIdentifierBlock) (NSString *kind, NSIndexPath * _Nullable indexPath);
```
Comment (flippinjoe21) [inline Pod/Classes/URBNDataSourceAdapter.h]: Is there a reason you're using `_Nonnull` and `_Nullable` when you use `nullable` below?

Comment (codecaffeine) [inline Pod/Classes/URBNDataSourceAdapter.h]: Yes, because blocks suck

Comment (flippinjoe21) [inline Pod/Classes/URBNDataSourceAdapter.h]: Fair enough

File: Pod/Classes/URBNAccordionDataSourceAdapter.h [hunk 2]
```diff
@@ -27,7 +29,7 @@ typedef void (^URBNAccordionHeaderViewConfigureBlock) (id view, id object, NSInt
  *  @param viewClass          The view class to configure.
  *  @param configurationBlock The block that configures instances of the cell class.
  */
-- (void)registerAccordionHeaderViewClass:(Class)viewClass withConfigurationBlock:(URBNAccordionHeaderViewConfigureBlock)configurationBlock;
+- (void)registerAccordionHeaderViewClass:(nullable Class)viewClass withConfigurationBlock:(URBNAccordionHeaderViewConfigureBlock)configurationBlock;
```
Comment (jgrandelli) [inline Pod/Classes/URBNAccordionDataSourceAdapter.h]: Do we really want someone to register a nil class here?

File: Pod/Classes/URBNAccordionDataSourceAdapter.h [hunk 3]
```diff
@@ -39,7 +41,7 @@ typedef void (^URBNAccordionHeaderViewConfigureBlock) (id view, id object, NSInt
  *  @param identifier (optional)    The reuseIdentifier to be used for this view.  If nil the @viewClass will be used.
  *  @param configurationBlock       The block that configures instances of the view class.
  */
-- (void)registerAccordionHeaderViewClass:(Class)viewClass withIdentifier:(NSString *)identifier withConfigurationBlock:(URBNAccordionHeaderViewConfigureBlock)configurationBlock;
+- (void)registerAccordionHeaderViewClass:(nullable Class)viewClass withIdentifier:(nullable NSString *)identifier withConfigurationBlock:(URBNAccordionHeaderViewConfigureBlock)configurationBlock;
```
Comment (jgrandelli) [inline Pod/Classes/URBNAccordionDataSourceAdapter.h]: i say again

File: Pod/Classes/URBNAccordionDataSourceAdapter.h (hunk 2 above)
Comment (codecaffeine) [inline Pod/Classes/URBNAccordionDataSourceAdapter.h]: The example code does. If you use a nib you don’t need to specify a class

File: Pod/Classes/URBNAccordionDataSourceAdapter.h (hunk 3 above)
Comment (codecaffeine) [inline Pod/Classes/URBNAccordionDataSourceAdapter.h]: I reply again. Check out the nib based code

File: Pod/Classes/URBNAccordionDataSourceAdapter.h (hunk 2 above)
Comment (jgrandelli) [inline Pod/Classes/URBNAccordionDataSourceAdapter.h]: #FairEnough

File: Pod/Classes/URBNAccordionDataSourceAdapter.h (hunk 3 above)
Comment (jgrandelli) [inline Pod/Classes/URBNAccordionDataSourceAdapter.h]: #FairEnough

Comment (flippinjoe21) [inline Pod/Classes/URBNAccordionDataSourceAdapter.h]: It's allowed to register a nil class incase you want to register an identifier from a prototypeCell

File: Pod/Classes/URBNDataSourceAdapter.h (hunk 1 above)
Comment (jgrandelli) [inline Pod/Classes/URBNDataSourceAdapter.h]: Do we think its possible to return a nil indexpath here?

File: Pod/Classes/URBNDataSourceAdapter.h [hunk 4]
```diff
@@ -9,16 +9,18 @@
...
+typedef NSString * _Nonnull (^URBNReuseableIdentifierBlock) (id _Nullable item, NSIndexPath * _Nullable indexPath);
 
-typedef void (^URBNCellConfigureBlock) (id cell, id object, NSIndexPath* indexPath);
-typedef void (^URBNSupplementaryViewConfigureBlock) (id view, URBNSupplementaryViewType kind, NSIndexPath* indexPath);
+typedef void (^URBNCellConfigureBlock) (id _Nullable cell, id _Nullable object, NSIndexPath *indexPath);
+typedef void (^URBNSupplementaryViewConfigureBlock) (id _Nullable view, URBNSupplementaryViewType kind, NSIndexPath * _Nullable indexPath);
```
Comment (jgrandelli) [inline Pod/Classes/URBNDataSourceAdapter.h]: Same

File: Pod/Classes/URBNDataSourceAdapter.h [hunk 5]
```diff
@@ -129,11 +131,11 @@ IB_DESIGNABLE @interface URBNDataSourceAdapter : NSObject <URBNDataSourceAdapter
...
  *  @param configurationBlock       The block that configures instances of the class
  */
-- (void)registerSupplementaryViewClass:(Class)viewClass ofKind:(URBNSupplementaryViewType)kind withIdentifier:(NSString *)identifier withConfigurationBlock:(URBNSupplementaryViewConfigureBlock)configurationBlock;
+- (void)registerSupplementaryViewClass:(nullable Class)viewClass ofKind:(URBNSupplementaryViewType)kind withIdentifier:(nullable NSString *)identifier withConfigurationBlock:(URBNSupplementaryViewConfigureBlock)configurationBlock;
 
-- (URBNSupplementaryViewConfigureBlock)viewConfigurationBlockForIdentifier:(NSString *)identifier withKind:(NSString *)kind;
+- (nullable URBNSupplementaryViewConfigureBlock)viewConfigurationBlockForIdentifier:(NSString *)identifier withKind:(NSString *)kind;
 
-- (NSString *)supplementaryIdentifierForType:(URBNSupplementaryViewType)type atIndexPath:(NSIndexPath *)indexPath;
+- (nullable NSString *)supplementaryIdentifierForType:(URBNSupplementaryViewType)type atIndexPath:(nullable NSIndexPath *)indexPath;
```
Comment (jgrandelli) [inline Pod/Classes/URBNDataSourceAdapter.h]: same

File: Pod/Classes/URBNDataSourceAdapter.h (hunk 1 above)
Comment (codecaffeine) [inline Pod/Classes/URBNDataSourceAdapter.h]: Ugh, nope. For some reason I though `[NSIndexPath indexPathForItem:0 inSection:section]` could return nil, but look at that beautiful `_Nonnull`:
screen shot 2015-11-02 at 4 22 11 pm

Comment (codecaffeine) [inline Pod/Classes/URBNDataSourceAdapter.h]: fixed

File: Pod/Classes/URBNDataSourceAdapter.h (hunk 4 above)
Comment (codecaffeine) [inline Pod/Classes/URBNDataSourceAdapter.h]: fixed

File: Pod/Classes/URBNDataSourceAdapter.h (hunk 5 above)
Comment (codecaffeine) [inline Pod/Classes/URBNDataSourceAdapter.h]: fixed

**Pull request merged**